R/reliabilityAnalysisBayesian.R

```diff
@@ -175,30 +175,28 @@ reliabilityBayesian <- function(jaspResults, dataset, options) {
 
       
       # add the scale info
-      corsamp <- apply(relyFit[["Bayes"]][["covsamp"]], c(1, 2), .cov2cor.callback, progressbarTick)
-      relyFit[["Bayes"]][["samp"]][["avg_cor"]] <- coda::mcmc(apply(corsamp, c(2, 3), function(x) mean(x[x!=1])))
+      relyFit[["Bayes"]][["samp"]][["avg_cor"]] <- matrix(0, dim(relyFit[["Bayes"]][["covsamp"]])[1], dim(relyFit[["Bayes"]][["covsamp"]])[2])
+      for (i in 1:dim(relyFit[["Bayes"]][["covsamp"]])[1]) {
+        for (j in 1:dim(relyFit[["Bayes"]][["covsamp"]])[2]) {
+          corm <- .cov2cor.callback(relyFit[["Bayes"]][["covsamp"]][i, j, , ], progressbarTick)
+          relyFit[["Bayes"]][["samp"]][["avg_cor"]][i, j] <- mean(corm[corm!=1])
+        }
+      }
```

Comment:
Using `nrow` and `ncol` is slightly more readable, (`nrow(x) = dim(x)[1L]`, likewise for `ncol(x)`).

Considering `mean(corm[corm!=1])`, what's the goal here? Compute the average correlation without the diagonal? If yes, this is not an ideal way of doing that because (1) limits of floating point comparison and (2) you're counting every value double (which doesn't change the result but is unnecessary). How about:

```suggestion
      lowerTriangleIndex = which(lower.tri(relyFit[["Bayes"]][["covsamp"]][i, j, , ]))
      for (i in 1:nrow(relyFit[["Bayes"]][["covsamp"]])) {
        for (j in 1:ncol(relyFit[["Bayes"]][["covsamp"]])) {
          corm <- .cov2cor.callback(relyFit[["Bayes"]][["covsamp"]][i, j, , ], progressbarTick)
          relyFit[["Bayes"]][["samp"]][["avg_cor"]][i, j] <- mean(corm[lowerTriangleIndex])
        }
      }
```
you could adapt `.cov2cor.callback` to handle lower triangular input as well but I don't think that's worth the effort.